caluma/caluma_workflow/api.py

```diff
@@ -100,3 +100,24 @@ def cancel_case(case: models.Case, user: BaseUser) -> models.Case:
     domain_logic.CancelCaseLogic.post_cancel(case, user)
 
     return case
+
+
+def cancel_work_item(work_item: models.WorkItem, user: BaseUser) -> models.WorkItem:
+    """
+    Cancel a work item (just like `CancelWorkItem`).
+
+    >>> cancel_work_item(
+    ...     work_item=models.WorkItem.first(),
+    ...     user=AnonymousUser()
+    ... )
+    <WorkItem: WorkItem object (some-uuid)>
+    """
+    domain_logic.CancelWorkItemLogic.validate_for_cancel(work_item)
+
+    validated_data = domain_logic.CancelWorkItemLogic.pre_cancel({}, user)
+
+    models.WorkItem.objects.filter(pk=work_item.pk).update(**validated_data)
```

Comment:
As described [here](https://django-simple-history.readthedocs.io/en/latest/common_issues.html#queryset-updates-with-history-updated-in-django-2-2), this doesn't work with `django-simple-history`. So you should iterate over the queryset.

Reply:
:astonished: We use this alot in this file. I'll open a PR to fix the existing usage of this..

Reply:
Fixed!